tests/test_smoke.py

```diff
@@ -2291,6 +2411,85 @@ def test_invalid_names(self, invalid_name_list, store_type):
                 storage_obj = storage_lib.Storage(name=name)
                 storage_obj.add_store(store_type)
 
+    @pytest.mark.parametrize(
+        'gitignore_structure, store_type',
+        [(GITIGNORE_SYNC_TEST_DIR_STRUCTURE, storage_lib.StoreType.S3),
+         (GITIGNORE_SYNC_TEST_DIR_STRUCTURE, storage_lib.StoreType.GCS),
+         pytest.param(GITIGNORE_SYNC_TEST_DIR_STRUCTURE,
+                      storage_lib.StoreType.R2,
+                      marks=pytest.mark.cloudflare)])
+    def test_excluded_file_cloud_storage_upload_copy(self, gitignore_structure,
+                                                     store_type):
+        # tests if files included in .gitignore and .git/info/exclude are
+        # excluded from being transferred to Storage
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # Create files to be uploaded in the Storage
+            self.create_dir_structure(tmpdir, gitignore_structure)
+
+            # Create files to be excluded from .gitignore and list them in the file
+            skypilot_path = os.path.abspath(sys.argv[0])
+            skypilot_path = os.path.dirname(skypilot_path)
```

Comment:
Can we get the path to skypilot repo using `a_module.__file__` instead?  Makes the test robust to invocations from outside of skypilot repo root.